File: seon/level3/12938.js
function solution(n, s) {
  if (s / n < 1) return [-1];

  const quotient = parseInt(s / n);
  const remainder = s % n;

  const answer = new Array(n).fill(quotient);
  for (let i = n - 1; i >= n - remainder; i--) {
    answer[i] += 1;
  }

  return answer;
}

File: eunna/level3/12938.py
def solution(n, s):
    answer = []
    
    if s < n:
        return [-1]
    
    num = s
    for i in range(n):
        quotient = s // n
        answer.append(quotient)
        num -= quotient
    
    if num == 0:
        return answer
    
    for i in range(num):
        answer[len(answer) - 1 - i] += 1
    
    return answer